src/main/java/prefs/Preferences.java

```diff
@@ -11,122 +17,262 @@
-        global = fileHandler.initGlobalConfig();
+    /**
+     * Saves the session and user configs to file
+     */
+    private void save() {
+        try {
+            saveConfig(sessionConfig, configDirectory, sessionConfigFileName, SessionConfig.class);
+
+        } catch (IOException e) {
+            HTLog.error(logger, e);
+            logger.error("Could not save session config");
+        }
+
+        try {
+            saveConfig(userConfig, configDirectory, userConfigFileName, UserConfig.class);
+
+        } catch (IOException e) {
+            HTLog.error(logger, e);
+            logger.error("Could not save user config");
+        }
     }
 
-    private void loadGlobalConfig() {
-        global = fileHandler.loadGlobalConfig();
+    /**
+     * Saves a Config to file
+     * @param config The config object to save.
+     * @param configDirectory The directory that the config file is held in.
+     * @param configFileName The file name of the config file on disk
+     * @param configClass The class of the config object
+     */
+    private void saveConfig(Config config,
+                            String configDirectory,
+                            String configFileName,
+                            Class configClass) throws IOException {
+        JsonHelper jsonHelper = new JsonHelper();
+        String jsonString = jsonHelper.toJsonString(config, configClass);
+        FileHelper fileHelper = new FileHelper(configDirectory, configFileName);
+        fileHelper.writeFileContents(jsonString);
     }
 
-    // Last login credentials. While the main UI is running (i.e. logged in successfully), last login
-    // credentials are guaranteed to be the current user's credentials thanks to setLastLoginCredentials
-    // being called immediately after a successful login in LoginDialog.
+    /**
+     * Last login credentials. While the main UI is running (i.e. logged in successfully), last login
+     * credentials are guaranteed to be the current user's credentials thanks to setLastLoginCredentials
+     * being called immediately after a successful login in LoginDialog.
+     */
     public String getLastLoginPassword() {
-        return global.getLastLoginPassword();
+        return sessionConfig.getLastLoginPassword();
     }
 
+    /**
+     * Retrieves the last login username
+     */
     public String getLastLoginUsername() {
-        return global.getLastLoginUsername();
+        return sessionConfig.getLastLoginUsername();
     }
 
+    /**
+     * Sets the last login credentials
+     * @param username The username used during the last login
+     * @param password The password used during the last login
+     */
     public void setLastLoginCredentials(String username, String password) {
-        global.setLastLoginCredentials(username, password);
+        sessionConfig.setLastLoginCredentials(username, password);
+        save();
     }
 
+    /**
+     * Retrieves a list of the last open filters
```

Comment:
This is confusing, but I don't want to touch renaming of methods irrelevant to this PR. Since you set the comment, maybe make it more explanative. `Retrieves a list of panel filters from last session`?